data/transactions/keyreg.go

```diff
@@ -27,11 +26,12 @@ import (
 type KeyregTxnFields struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
-	VotePK          crypto.OneTimeSignatureVerifier `codec:"votekey"`
-	SelectionPK     crypto.VRFVerifier              `codec:"selkey"`
-	VoteFirst       basics.Round                    `codec:"votefst"`
-	VoteLast        basics.Round                    `codec:"votelst"`
-	VoteKeyDilution uint64                          `codec:"votekd"`
+	VotePK           crypto.OneTimeSignatureVerifier `codec:"votekey"`
+	SelectionPK      crypto.VRFVerifier              `codec:"selkey"`
+	VoteFirst        basics.Round                    `codec:"votefst"`
+	VoteLast         basics.Round                    `codec:"votelst"`
+	VoteKeyDilution  uint64                          `codec:"votekd"`
+	Nonparticipation bool                            `codec:"nonpart"`
```

Comment:
Really would have made sense to make this a separate txn type.  Overloading an almost entirely orthogonal txn for this is a bad idea.

Reply:
I don't think it's orthogonal -- keyreg transactions are for changing account status. Now they can  change status to NonParticipating in addition to Online and Offline.

Reply:
This situation is similar to keyreg transactions that register an empty key, where we interpret it semantically as a different case. Whether that's good design or not, I'm ambivalent, but at least there is already a precedent in the code base.